packages/components/src/components/post-collapsible-trigger/post-collapsible-trigger.tsx

```diff
@@ -43,9 +43,12 @@ export class PostCollapsibleTrigger {
   componentWillLoad() {
     this.root = getRoot(this.host);
 
-    this.root.addEventListener('postToggle', (e: PostCollapsibleCustomEvent<boolean>) => {
-      if (!this.trigger || !e.target.isEqualNode(this.collapsible)) return;
-      this.trigger.setAttribute('aria-expanded', `${e.detail}`);
+    this.root.addEventListener('postToggle', (e: CustomEvent) => {
+      eventGuard.call(this, e, { targetLocalName: 'post-collapsible' }, () => {
+        if (this.trigger) {
+          this.trigger.setAttribute('aria-expanded', `${e.detail}`);
+        }
+      });
     });
   }
```

Comment:
Switch hook from componentWillLoad to connectedCallback.
Make sure, you remove the registered eventlistener in the disconnectedCallback.
And by the way, arrow/inline functions, can not be used if you properly want to remove an eventlistener later.
Instead, pass a private function defined in the class it self as the callback function parameter of the addEventListener.

```ts
class bla {
  connectedCalllback() {
    this.root = getRoot(this.host);

    this.root.addEventListener('postToggle', this.toggleHandler);
  }

  disconnectedCallback() {
    this.root.removeEventListener('postToggle', this.toggleHandler);
  }
  
  private toggleHandler(e: CustomEvent) {
     ...
  }
}
```